src/iosource/Manager.cc

```diff
@@ -175,6 +175,14 @@ void Manager::FindReadySources(ReadySources* ready)
 					}
```

Comment:
If we're going to just automatically insert zero-timeout sources, I think this whole if statement could change as well to something like:

```
if (timeout == -1 || (next > 0 && next < timeout))
  {
  timeout = next;
  timeout = iosource;
  }
```

There's no reason to check for the zero-timeout case because we're covering it already later. I'm slightly concerned that a zero-timeout source could lead to starvation of any other timeout sources, but I guess they'd eventually fire when they hit the case where their timeout is zero as well.

Reply:
As in DM, tried to adapt with your suggestion. Not sure it's better now.

I think the IsLive() special case could be stream-lined a bit, but not wanting to change too much at once...